data/rancher/ui_test.py

```diff
@@ -0,0 +1,45 @@
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+import time
+
+if __name__ == '__main__' :
+  driver = webdriver.Firefox()
+  wait=WebDriverWait(driver, 5)
+
+  driver.get("https://localhost:10443")
+  time.sleep(2)
+
+  # Check page started
+  assert "Rancher" in driver.title
+
+  # Initial page: Change password page
+  wait.until(EC.visibility_of_element_located((By.XPATH, "//input[@type='password']")))
+  inputs = driver.find_elements_by_xpath("//input[@type='password']")
+
+  for input in inputs:
+    input.send_keys("123456asdfgh")
+
+  checkboxes = driver.find_elements_by_xpath("//input[@type='checkbox']")
+  checkboxes[1].click()
+
+  driver.find_element_by_xpath("//button[@type='submit']").click()
+
+  # Initial page: set the URL
+  time.sleep(2)
```

Comment:
You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted.

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

Same applies for other uses of sleep

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details